test/provision_test.py

```diff
@@ -0,0 +1,185 @@
+import json
+from test.test_util import TestUtils
+import pytest
+
+from autoprotocol.protocol import Protocol
+
+
+class TestProvision(object):
+    p = Protocol()
+    w1 = (
+        p.ref("w1", None, cont_type="96-pcr", discard=True)
+        .well(0)
+        .set_volume("2:microliter")
+    )
+
```

Comment:
you should make this a pytest fixture as your tests _could_ break, if run out of order. Additionally, the `Protocol` object is mutable with side-effecting methods. This will also remove redundant boilerplate if you declare your protocol, container, and well objects upfront, instead of declaring them every time.